### ISSUE TITLE
Ultimate Psionics-  Life Leech Archetype

### ISSUE THREAD

**Issue by JerrinStrongblade** [shadowchemosh/HL-Pack-3PP#17]
I'm trying to convert my Vitalist Life leech/soul thief to HL.   However, it looks like Life leech has not been included in HL UP.   Could you please bump this up in priority?

**Comment by psych777** [comment]
was trying to finalize this archetype and have come across a problem. at 2nd level it gets a class special which gives a bonus feat. so i've got it listed to recieve the special at level 2, which then is bootstrapped with the feat, but as soon as i add the archetype at level 1, it is granting the bonus feat. do i need some other kind of prereq for it not to take the bonus feat until level 2?

**Comment by shadowchemosh** [comment]
You have to use a bootstrap condition to control when the feat will appear. Take a look at the Rangers class special endurance for an example of how to code this.

**Comment by psych777** [comment]
so the ranger's has:
      (count:Classes.Ranger >= 3 | count:Classes.KPSpLesRan >= 3 | count:Classes.KPSkinCha >= 3) & !AbReplace.cRgrEndur

assume this would be (count:Classes.Vitalist >=2) for me, but what does the !AbReplace.cRgrEndur do exactly?

**Comment by shadowchemosh** [comment]
The check for !AbReplace.cRgrEndur is to help support Archetypes.  So if the class Ability cRgrEndur gets replaced a tag is placed on the hero (AbReplace.cRgrEndur).  So the "!" is saying "NOT".  So in plain English it reads  "If Ranger Level 3+ AND We have NOT had our Class Ability cRgrEndur Replaced" then go ahead and give the feat.

So our bootstrap condition needs to say
count:Classes.Vitalist >=2 & !AbReplace.XXXXXXXX  where XXXXXXX is the unique ID of your Archetype Class Special.

**Comment by shadowchemosh** [comment]
Another "FYI".  Class Specials and Racial Specials should NOT be "Source ID" marked.  The Thing they are bootstrapped too should be Source ID marked.  This way a class ability or racial ability could be reused by another Class or Race without requiring UP to be active.

**Comment by psych777** [comment]
so we don't want the source of DSP Ultimate Psionics on them? but we still want the product identity for hero lab community?
also, i just noticed after making pulls that changes i made dont seem to be in there...looks like all the changes i recently pushed up may be gone...this happened before too...why is this happening?

Date: Sat, 28 Mar 2015 13:08:18 -0700

**Comment by shadowchemosh** [comment]
"so we don't want the source of DSP Ultimate Psionics on them? but we still want the product identity for hero lab community?"
--Correct.

"also, i just noticed after making pulls that changes i made dont seem to be in there...looks like all the changes i recently pushed up may be gone...this happened before too...why is this happening?"
A) double check once you do a fresh pull down of the repository. I have noticed things can look funky until that is done.

b) Make sure you are doing a "Push" not just a commit.  A commit only happens on your personal computer.  It requires a 'push' to actually get to the repository. =)

**Comment by psych777** [comment]
my normal pattern is to commit then push. in the ss you just pushed a couple so i'm 2 behind. if i'm behind i always pull first so i don't overwrite someone else's work. so is there any way to get the changes from that commit to come back (without manually making all the changes again)?
![image](https://cloud.githubusercontent.com/assets/10572815/6882871/5972abea-d58c-11e4-9893-a2bd731149d2.png)
and while looking at that screenshot, something else that always made me wonder...in the bottom right you can see it is saying lines 162 and 163 were deleted (at least i assume that is the '-' meaning), but i never touched those fields.

**Comment by shadowchemosh** [comment]
I always commit and push in one action.  There is a check box below where you put in a comment that says "Push Changes immediately to origin/master".  But doing a Push should do any commits you have saved.

You will get changes to the XML as you make changes.  So what is saying is that <fieldval elements must be BEFORE <arrayval elements.  So that is what the HL editor did.  Git sees it as a delete/add.  It really is just a insert.  But its not how it is seen.

What changes did you have?  You sure they are gone?  I have only seen issues where Git finds two people have changed the exact same line of code.  Then it asks you to manually adjust the XML to the final version.

**Comment by psych777** [comment]
i had added the source page and book to all the classes and archetypes in PU, added some feats, updated feats A-C...fixed the two feats that had issues reported...its all gone when i go in the editor and look now

**Comment by shadowchemosh** [comment]
I am sorry I don't have any good ideas.  Give a try doing a straight "push" from the top tool bar.  If anything is "saved" (ie committed) it should push those changes right away.

Only thing else I can say is "save" and push your changes often.  I do a specific "set" of changes and then push to git.  I don't wait long periods of time before doing that.  So in example as soon as you set all the Book Info I would set my comment and push to git.  Then as soon as I added the new feats push and comment.  Then make the misc changes to the feats.

**Comment by shadowchemosh** [comment]
![image1](https://cloud.githubusercontent.com/assets/7717785/6883010/1910f9e4-d569-11e4-98ad-fa535b7c5fe2.jpg)
Only thing that just came to me.  When you do the PULL make sure you have the "Commit Merged Changes Immediately" check marked.

**Comment by psych777** [comment]
i made the majority of the changes last night and swear i pushed them. i do have that check box checked on my pull menu.

**Comment by shadowchemosh** [comment]
Ummm... I am sorry I don't know.  I have not had that happen before. =(

**Comment by psych777** [comment]
okay, i found a way to get my files back. before i proceed wanted to ask your advice on how to proceed.
so you can see my 'master' is 3 steps back of where the current project is on github. i copied in my two files i had edited.
in two of those three steps you have edited the same file (UP Classes), and one of them you edited the other file i have (UP Feats). 
so to incorporate my edits ("uncommitted changes" in the screenshot) without deleting yours, should i pull first and then commit/push? or should i commit, then pull, then push?
![image](https://cloud.githubusercontent.com/assets/10572815/6883189/efab4e70-d599-11e4-8bd5-7cf7c8c83edd.png)

**Comment by shadowchemosh** [comment]
I would copy the Classes/Feats to your desktop just to be safe.  Then what should happen is you do a the "pull" with merge.  That should merge my changes into your files.  As long as I didn't change the exact same thing as you (maybe) it will merge in without issues.  Then you can push the now "merged" file back to github.

If the merge is not 100% successfull Git will leave the file in a state of half merge. HL will NOT load and in the XML will be "<<<<<<<<<<<<<<<<<" characters marking the file for where a human needs to go into the XML and fix it.

If the above happens I will be happy to fix the merged file.  Its partially my fault anyways for playing in the same area.  So email me the file and I can fix and send it back to you.  email shadowchemosh@yahoo.com. =)

**Comment by psych777** [comment]
okay. i thought i had everything finally on track. it all looked good but then herolab wouldn't load and on the line it errored it did have a <<<<<<<<<<<<<<< HEAD
so i did a reverse commit and now herolab loads again, just without my changes. i will email you the 2 files...just not sure what to do in sourcetree now as it shows my master on the commit that caused problems:
![image](https://cloud.githubusercontent.com/assets/10572815/6883418/16b96984-d5a4-11e4-81a4-d27da87e70f2.png)

**Comment by shadowchemosh** [comment]
I am fixing the merge issues now.  I am thinking you should not have done a reverse commit.  So I should have the files fixed shortly and will push to git.

**Comment by shadowchemosh** [comment]
I have the files fixed and pushed back to git.  Maybe discard what you have and pull down the latest files.

It should now have both my changes and yours living together. =)

**Comment by psych777** [comment]
so pulling your new files gave me the result of the screenshot 1. looking at the file status, if i just delete the lines in xml pointed to by the red arrows in ss's 2 and 3, would that fix things up for me?if not, can you email me the classes/feats files you have so i can put them in my folder and then it should match up..

Date: Sat, 28 Mar 2015 16:54:58 -0700

**Comment by psych777** [comment]
just realized i replied to a github email instead of your normal email so here are the screenshots
![1](https://cloud.githubusercontent.com/assets/10572815/6883599/93dd8622-d5ac-11e4-9768-af519b957ce8.jpg)
![2](https://cloud.githubusercontent.com/assets/10572815/6883600/9a63e7a2-d5ac-11e4-85a5-a40dacddda73.jpg)
![3](https://cloud.githubusercontent.com/assets/10572815/6883601/a30420ac-d5ac-11e4-9eaa-e78f47e06a2b.jpg)

**Comment by shadowchemosh** [comment]
If you click on the file on github website then click on RAW you can download the current file directly from github and over write your file.

**Comment by psych777** [comment]
ok, finally have my tree looking good. hopefully nothing has been messed up!

**Comment by psych777** [comment]
archetype is in. anything further will be a bug report

**Comment by JerrinStrongblade** [comment]
Thank you for taking the time to add this.   I appreciate it.
Jerrin


     On Sunday, June 14, 2015 1:24 PM, psych777 <notifications@github.com> wrote:
   

 archetype is in. anything further will be a bug report—
